### PR TITLE
Add Ethernet Global Data support

### PR DESCRIPTION
**Pull request by 0xA50C1A1**
Please sign (check) the below before submitting the Pull Request:

- [x] I have signed the ntop Contributor License Agreement at https://github.com/ntop/legal/blob/main/individual-contributor-licence-agreement.md
- [x] I have read the contributing guide lines at https://github.com/ntop/nDPI/blob/dev/CONTRIBUTING.md
- [x] I have updated the documentation (in doc/) to reflect the changes made (if applicable)

Describe changes:

Ethernet Global Data is an industrial network protocol that utilizes UDP for communication. It enables real-time data transfer and communication between industrial devices and controllers over Ethernet networks.

### CODE REVIEW

File: src/lib/protocols/egd.c [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+  NDPI_EXCLUDE_PROTO(ndpi_struct, flow);
+}
+
+void init_egd_dissector(struct ndpi_detection_module_struct *ndpi_struct,u_int32_t *id)
+{
+  ndpi_set_bitmask_protocol_detection("EthernetGlobalData", ndpi_struct, *id,
+                                      NDPI_PROTOCOL_EGD,
+                                      ndpi_search_egd,
+                                      NDPI_SELECTION_BITMASK_PROTOCOL_UDP_WITH_PAYLOAD,
```
Comment (IvanNardi) [inline src/lib/protocols/egd.c:60]: Could you add an explicit comment about the fact that this protocol seems to be ipv4 only, please?

File: tests/cfgs/default/result/h323-overflow.pcap.out [hunk 2]
```diff
@@ -13,18 +15,15 @@ Automa domain:        0/0 (search/found)
...
-
-Unrated                          1 58            1            
-
+HTTP	1	58	1
 
+Acceptable                       1 58            1            
 
-Undetected flows:
-	1	TCP 192.168.1.1:31337 -> 192.168.1.2:80 [proto: 0/Unknown][IP: 0/Unknown][ClearText][Confidence: nBPF][DPI packets: 1][1 pkts/58 bytes -> 0 pkts/0 bytes][Goodput ratio: 7/0][< 1 sec][Plen Bins: 100,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
+	1	TCP 192.168.1.1:31337 -> 192.168.1.2:80 [proto: 7/HTTP][IP: 0/Unknown][ClearText][Confidence: Match by port][DPI packets: 1][cat: Web/5][1 pkts/58 bytes -> 0 pkts/0 bytes][Goodput ratio: 7/0][< 1 sec][Risk: ** Unidirectional Traffic **][Risk Score: 10][Risk Info: No server to client traffic][Plen Bins: 100,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
```
Comment (IvanNardi) [inline tests/cfgs/default/result/h323-overflow.pcap.out]: This change seems wrong... could you check, please?

**Review by IvanNardi — CHANGES_REQUESTED** [review]

File: tests/cfgs/default/result/h323-overflow.pcap.out (hunk 2 above)
Comment (0xA50C1A1) [reply tests/cfgs/default/result/h323-overflow.pcap.out]: > This change seems wrong... could you check, please?

Yea, it's really weird

**Review by IvanNardi — APPROVED** [review]
Thank you

**Pull request merged**